### DISCUSSION TITLE
Don't add semicolon at ever ctrl + s

### DISCUSSION THREAD

**Discussion by supersuryaansh** [zed-industries/zed#18569] [Feature Requests]
Every time I hit save, zed formats my code which I want it to do, but it also adds semicolon after every line of my js code, which I do not want it to do.

Is there a way to alter this behaviour?

**Comment by omarjatoi** [comment]
This behavior is the default in the `typescript-language-server`. You could either configure Zed to use and external tool like `eslint` or `prettier` and add configuration for that in your project (https://zed.dev/docs/languages/javascript#code-formatting). FWIW, this is what I would recommend.

Or you could try to configure the `typescript-language-server` using the [`javascript.format.semicolons: 'remove'` option](https://github.com/typescript-language-server/typescript-language-server/blob/master/docs/configuration.md#workspacedidchangeconfiguration). [This section in Zed documentation]( https://zed.dev/docs/languages/typescript#inlay-hints) might be helpful setting an override for semicolons but full disclosure I'm not sure you _can_ set those workspace options.

**Reply by martoche** [reply]
Thanks! Here is what I've added to my `settings.json` file:
```json
  "lsp": {
    "vtsls": {
      "settings": {
        "typescript": {
          "format": {
            "semicolons": "remove"
          }
        }
      }
    }
  },
```

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links

**Comment by devkral** [comment]
the problem is prettier. You can configure it with prettier: false like in typescript.
vtsls is completely unrelated

```jsonc
...
"languages": {
  "JavaScript": {"prettier": false}
}
...
```
